### PR TITLE
Change Adhese bidder to use /openrtb2 endpoint

### PR DESCRIPTION
**Pull request by karol-adhese**
- switch endpoint and therefore adjust the payload manipulation, adjust tests
- Go version of the server does not have the same changes, PR for those should follow

### CODE REVIEW

**Review by JakubKAdhese — APPROVED** [review]

File: src/main/java/org/prebid/server/bidder/adhese/AdheseBidder.java [hunk 1]
```diff
@@ -109,30 +101,25 @@ private ExtImpAdhese parseImpExt(Imp imp) {
         }
     }
 
-    private AdheseRequestBody buildBody(BidRequest request, ExtImpAdhese extImpAdhese) {
+    private String getUrl(ExtImpAdhese extImpAdhese) {
+        return endpointUrl.replace("{{AccountId}}", extImpAdhese.getAccount());
+    }
+
+    private BidRequest modifyBidRequest(BidRequest bidRequest, ExtImpAdhese extImpAdhese) {
```
Comment (And1sS) [inline src/main/java/org/prebid/server/bidder/adhese/AdheseBidder.java:108]: Please, place methods in call order.

File: src/main/java/org/prebid/server/bidder/adhese/AdheseBidder.java [hunk 2]
```diff
@@ -109,30 +101,25 @@ private ExtImpAdhese parseImpExt(Imp imp) {
...
-                .parameters(parameterMap)
-                .build();
-    }
+        final ObjectNode adheseExtInnerNode = mapper.mapper().valueToTree(parameterMap);
+        final ObjectNode adheseExtNode = mapper.mapper().createObjectNode().set("adhese", adheseExtInnerNode);
 
-    private String getUrl(ExtImpAdhese extImpAdhese) {
-        return endpointUrl.replace("{{AccountId}}", extImpAdhese.getAccount());
-    }
+        Imp imp = bidRequest.getImp().get(0).toBuilder()
```
Comment (And1sS) [inline src/main/java/org/prebid/server/bidder/adhese/AdheseBidder.java:116]: Should be final, also please, check for similar occurences.

File: src/main/java/org/prebid/server/bidder/adhese/AdheseBidder.java [hunk 3]
```diff
@@ -146,96 +133,44 @@ private Map<String, List<String>> parseTargetParametersAndSort(JsonNode targets)
...
+        if (bidResponse == null || bidResponse.getSeatbid() == null) {
+            return Optional.empty();
+        }
 
-    private static Map<String, List<String>> getIfaParameter(Device device) {
-        final String ifa = device != null ? device.getIfa() : null;
-        return StringUtils.isNotBlank(ifa)
-                ? Collections.singletonMap(IFA_QUERY_PARAMETER, Collections.singletonList(ifa))
-                : Collections.emptyMap();
+        return bidResponse.getSeatbid().stream()
```
Comment (And1sS) [inline src/main/java/org/prebid/server/bidder/adhese/AdheseBidder.java:148]: Can be simplified to smth like:
```java
return Optional.ofNullable(bidResponse)
    .map(BidResponse::getSeatbid)
    .stream()
    .flatMap(Collection::stream)
    .map(SeatBid::getBid)
    .filter(Objects::nonNull)
    .flatMap(Collection::stream)
    .findFirst();
```

**Review by And1sS — CHANGES_REQUESTED** [review]

**Comment by karol-adhese** [comment]
I made the requested changes (in separate commits), please verify again

**Review by And1sS — APPROVED** [review]

**Pull request merged**